src/anyio/from_thread.py

```diff
@@ -116,7 +116,8 @@ async def run_async_cm(self) -> bool | None:
             # `_BlockingAsyncContextManager.__exit__` is called, and an
             # `_exit_exc_info` has been set.
             result = await self._async_cm.__aexit__(*self._exit_exc_info)
-            return result
+
+        return result
```

Comment:
Do you have a test for this?

Reply:
As mentioned on Gitter, these cases are indeed covered by tests.